init.d/urandom.in

```diff
@@ -21,33 +23,35 @@ depend()
 
 save_seed()
 {
-	local psz=1
-
-	if [ -e /proc/sys/kernel/random/poolsize ]; then
-		: $(( psz = $(cat /proc/sys/kernel/random/poolsize) / 4096 ))
-	fi
-
 	(	# sub shell to prevent umask pollution
 		umask 077
-		dd if=/dev/urandom of="$urandom_seed" count=${psz} 2>/dev/null
+		dd if=/dev/urandom of="$urandom_seed" count=1 2>/dev/null
 	)
 }
 
 start()
 {
-	[ -c /dev/urandom ] || return
-	if [ -f "$urandom_seed" ]; then
-		ebegin "Initializing random number generator"
-		cat "$urandom_seed" > /dev/urandom
-		eend $? "Error initializing random number generator"
+	if [ "$RC_UNAME" = Linux ]; then
+		seedrng
+	else
+		[ -c /dev/urandom ] || return
+		if [ -f "$urandom_seed" ]; then
+			ebegin "Initializing random number generator"
+			cat "$urandom_seed" > /dev/urandom
+			eend $? "Error initializing random number generator"
+		fi
+		rm -f "$urandom_seed" && save_seed
 	fi
-	rm -f "$urandom_seed" && save_seed
 	return 0
 }
 
 stop()
 {
-	ebegin "Saving random seed"
-	save_seed
-	eend $? "Failed to save random seed"
+	if [ "$RC_UNAME" = Linux ]; then
+		seedrng
+	else
+		ebegin "Saving random seed"
+		save_seed
+		eend $? "Failed to save random seed"
+	fi
```

Comment:
Might be considered nitpicking but can you please try to use curly braces in all variables? That would improve at least readability quite a bit:
`$urandom_seed` -> `${urandom_seed}`
`$RC_UNAME` -> `${RC_UNAME}`

Reply:
yes, absolutely none of OpenRC uses this, and I don't think this is the time to nitpick